pkg/operator/controller/controller.go

```diff
@@ -93,8 +95,15 @@ func newReconciler(mgr manager.Manager) (*ReconcileCDI, error) {
 
 	log.Info("", "VARS", fmt.Sprintf("%+v", namespacedArgs))
 
+	cfg := mgr.GetConfig()
+	cdiClient, err := cdiClientset.NewForConfig(cfg)
+	if err != nil {
+		log.Error(err, "Error building cdi clientset")
+	}
+
 	r := &ReconcileCDI{
 		client:         mgr.GetClient(),
+		cdiClientset:   cdiClient,
```

Comment:
You don't need the cdiClientSet, the runtime library client will register the CDI types, and you can use that in the operator. We currently have a duality with using 2 separate clients:
1. client-go (which requirest the cdiClientset
2. controller runtime-library (used in operator, cdiconfig, and I have several PRs pending to convert the other controllers)

So we are converging on 2, and thus I would encourage you to use the runtime library client instead of the client-go one. This will also safe you from having to create a fake cdiClientSet in the tests.

Reply:
Yeah this comment got me thinking.  Turns out that it is not possible to list all datavolumes in all namespaces with the operator client.  As of #1091 It has a cache bound to the install namespace.  This broke something.  But I fixed it in #1106.  Maya should use the uncached client created in that PR.